### ISSUE TITLE
Break the Doorbots code into multiple daemons

### ISSUE THREAD

**Issue by johnmckerrell** [DoESLiverpool/somebody-should#915]
Currently the `logcards.rb` script does a lot of things. It would be more resilient if it could only do the necessary things and have the less important things handled elsewhere. We could have a queue service running on the Pi which gets informed of a scan, then the audio player could respond to this one way, the Slack bot another way, the SSH-to-a-kindle bot another way without them causing the doors to not open.

We could also have the script shout out to an external queue, like the MQTT broker, but we have to be careful of privacy as we shouldn't be making every visit to the space public, and definitely shouldn't make the card IDs public. With this MQTT broker the audio playing could actually be handled by a completely separate system.

**Comment by notorious-gay** [comment]
Is this still needing to be done? I can have a crack if it is.

**Comment by johnmckerrell** [comment]
Yes, still needs doing. As it says above I think you’d want some local queue that could pass the card id around and then perhaps another daemon would read the ID and then emit the person’s nickname and/or audio file name to MQTT.local

**Comment by notorious-gay** [comment]
No worries, I'll take a look over the next week when I'm not in classes 👍🏻

**Comment by notorious-gay** [comment]
Sorry for how long I've taken to get back to you on this, I've been picking apart the script and playing around with it around assignments. I think I have a pretty good understanding of how the script interacts with the various bits of hardware.

With how the script is currently laid out, I'd like to begin with some smaller changes to PR over to document and break up the code so it is easier to read and then build some unit tests, then move on to breaking it up into a group of services. If that's agreeable for you?
